075-auto-rebalancing-cluster-scaling.md

```diff
@@ -0,0 +1,507 @@
+Another option could be leaving both the ways living together.
+If this proposal is considered valid, doing such a change on the `KafkaRebalance` custom resource would help the proposal itself because the operator should not copy over the entire configuration from the "template" resource to the actual rebalance resource but just set the configuration reference.
+Of course, such a change would need a different proposal to be approved before this one.
+
+It is also worth mentioning that a "template" `KafkaRebalance` custom resource doesn't add much complexity to the operator in terms of watching, reacting and processing.
+The `KafkaRebalanceAssemblyOperator` is just going to ignore such a resource when it's created, updated or deleted.
+It can be only referenced by an "actual" `KafkaRebalance` custom resource so its creation doesn't trigger any action by the operator.
+The operator only reads the "template" `KafkaRebalance` custom resource and generates a new corresponding "actual" `KafkaRebalance` custom resource when the cluster is being scaled.
+
+### Extending the `Kafka.status` field with auto-rebalancing related state
+
+The auto-rebalancing, together with the scaling action, can take several reconciliations to finish.
+For this reason, taking into account cluster operator might be restarted as well, we need to save some information about the state of the auto-rebalancing operation.
+The `Kafka` custom resource is extended by adding a new `status.autoRebalance` field.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+  cruiseControl:
+    # ...
+    autoRebalance:
+      # ...
+status:
+  autoRebalance:
+    lastTransitionTime: # timestamp of last transition during the auto-rebalancing flow
+    state: # if a scaling operation is running or what's the state of the current rebalancing
+    modes:
+      - mode: add-brokers
+        brokers: # list with brokers' IDs to add (as part of the scaling up)
+      - mode: remove-brokers
+        brokers: # list with brokers' IDs to remove (as part of the scaling down)
+```
+
+The `state` field provides the current state of the rebalancing process.
+Its value is defined by a Finite State Machine (FSM) described in the next sections which evolves even based on the status of the corresponding "actual" `KafkaRebalance` custom resource that was created for that purpose. 
+The `lastTransitionTime` field provides the timestamp of the latest auto-rebalancing state update.
+The `modes` field lists the enabled rebalancing modes, as defined through the `spec.cruiseControl.autoRebalance.mode` field, with the corresponding `brokers` list containing the brokers' IDs that are part of the scaling (up/down) actions to be useful across reconciliations.
```

Comment:
```suggestion

```
This can be removed now right, since it is described on the next line differently?